uint8 INVALID=0
uint8 SK_PARAMS_REQ=1
uint8 IK_PARAMS_REQ=2
uint8 type
uint8 id
